src/tanxconnector/client.py

```diff
@@ -481,120 +483,222 @@ def get_network_config(self):
+
+        Returns:
+            dict: A dictionary containing the status of the approval operation and the transaction hash if successful.
+        """
 
-        current_coin = filter_cross_chain_coin(polygon_config, coin, 'DEPOSIT')
+        # Fetch network configuration
+        network_config = self.get_network_config()
+        cross_network_config = network_config[network]
+        
+        # Retrieve contract address and relevant coin information
+        contract_address = cross_network_config['deposit_contract']
+        current_coin = filter_cross_chain_coin(cross_network_config, coin, 'DEPOSIT')
+        
+        # Extract token contract address
         token_contract = current_coin['token_contract']
+        
+        # Call utility function to approve unlimited allowance
         res = approve_unlimited_allowance_util(contract_address=contract_address, token_contract=token_contract, signer=signer, w3=w3)
+        
+        # Prepare and return response
+        res = {"status": "success", "payload": res}
+    
         return res
 
-    def deposit_from_polygon_network_with_signer(self, signer: Account, provider: Web3, currency:str, amount: float):
-        if float(amount)<=0:
+    
+    def deposit_from_polygon_network_with_signer(self, signer: Account, provider: Web3, currency:str, amount: float, **kargs):
+        self.deposit_from_cross_network_with_signer(signer, provider, currency, amount, network="POLYGON", **kargs)
+        return None
+    
+    def deposit_from_cross_network_with_signer(self, signer: Account, provider: Web3, currency: str, amount: float, network: str, **kwargs):
+        """
+            Deposit funds from a cross-chain network to the platform using the provided signer and provider.
+
+            Args:
+                signer (Account): The signer object used to sign the transaction.
+                provider (Web3): The Web3 provider object used to interact with the blockchain.
+                currency (str): The currency symbol of the funds being deposited.
+                amount (float): The amount of funds being deposited, in the same unit as the currency.
+                network (str): The network to which the funds are being deposited.
+                **kargs: Additional keyword arguments that can be used to specify gas options for the transaction.
+
+            Returns:
+                dict: A dictionary containing the status of the deposit operation and the transaction hash if successful.
+
+            Raises:
+                InvalidAmountError: If the amount to be deposited is less than or equal to zero.
+                BalanceTooLowError: If the balance of the user's account is insufficient for the deposit operation.
+                AllowanceTooLowError: If the allowance for the specified token is insufficient for the deposit operation.
+
+        """
+
+        # Check if amount is valid
+        if float(amount) <= 0:
```

Comment:
use `Decimal` instead of float

Reply:
`float` is being used through out the library, changing it to decimal would require to make changes throughout the library, which might break something, would require additional testing